packages/core/src/external/sftp/surescripts/schema/shared.ts

```diff
@@ -0,0 +1,250 @@
+// Wraps an SFTP file operation with a schema for each line of the generated file.
+export type FileFieldSchema<T extends object> = {
+  [K in keyof T]: FileField<T, K>;
+}[keyof T][];
+
+export type OutgoingFileRowSchema<T extends object> = {
+  [K in keyof T]: OutgoingFileField<T, K>;
+}[keyof T][];
+
+export type IncomingFileRowSchema<T extends object> = {
+  [K in keyof T]: IncomingFileField<T, K>;
+}[keyof T][];
+
+export type FileRowValidator<T extends object> = (row: object) => row is T;
+
+export interface OutgoingFileSchema<H extends object, D extends object, F extends object> {
+  header: OutgoingFileRowSchema<H>;
+  detail: OutgoingFileRowSchema<D>;
+  footer: OutgoingFileRowSchema<F>;
+}
+
+export interface FileValidator<H extends object, D extends object, F extends object> {
+  header: FileRowValidator<H>;
+  detail: FileRowValidator<D>;
+  footer: FileRowValidator<F>;
+}
+
+// Describes a single field for a row in a generated file.
+export interface FileField<T extends object, K extends keyof T = keyof T> {
+  field: number;
+  key?: K;
+  toSurescripts?: (row: T) => string;
+  fromSurescripts?: (value: string) => T[K];
```

Comment:
These are defined in the incoming/outgoing, do we really need them here?

Reply:
Yes these should have been marked as deprecated, and will be removed (I replaced them with an incoming/outgoing version of the interface with required conversion functions).